corporate/lib/stripe.py

```diff
@@ -1834,14 +1961,23 @@ def make_end_of_cycle_updates_if_needed(
             return None, None
         return None, last_ledger_entry
 
-    def get_next_plan(self, plan: CustomerPlan) -> Optional[CustomerPlan]:
+    def get_next_plan(self, plan: CustomerPlan) -> Union[CustomerPlan, CustomerPlanOffer, None]:
+        customer = plan.customer
         if plan.status == CustomerPlan.SWITCH_PLAN_TIER_AT_PLAN_END:
             assert plan.end_date is not None
             return CustomerPlan.objects.filter(
-                customer=plan.customer,
+                customer=customer,
                 billing_cycle_anchor=plan.end_date,
                 status=CustomerPlan.NEVER_STARTED,
             ).first()
+        elif customer.required_plan_tier is not None:
+            # Currently, the only case when a next_plan is scheduled with the
+            # current_plan.status NOT SET to SWITCH_PLAN_TIER_AT_PLAN_END is a
+            # fixed price plan configured via /support which the customer is yet
+            # to buy or schedule a purchase.
+            return get_configured_fixed_price_plan_offer(
+                customer=customer, plan_tier=customer.required_plan_tier
+            )
```

Comment:
Hmm. I don't think this change is correct; we do want to display these on the support panel, but `get_billing_page_context` probably shouldn't show this as a scheduled plan if the customer hasn't approved it? But it's a weird case that mostly shouldn't happen; I don't know if an assert is right or something else; we can discuss in chat.zulip.org and I think is not a blocker for this PR.

Reply:
It was correct because on billing page, if a plan is scheduled, then current plan would have a status=switch_plan_tier_at_end..., so it would never show configured CustomerPlanOffer.

Anyways, we changed that in #28763 -- 2nd commit due to different reasons.